tuxemon/map_loader.py

```diff
@@ -379,9 +387,12 @@ def load_event(
         w = int(obj.width / tile_size[0])
         h = int(obj.height / tile_size[1])
 
+        properties = cast(Dict[str, Any], obj.properties)
```

Comment:
I would prefer not to cast. Instead, we should properly type `pytmx`.

Reply:
Agreed

Reply:
What about natsorted? Their typehints are also bad.

Reply:
Well, I said that about `pytmx` because its author is a maintainer of this project, so they will certainly be accepted. If we can improve `natsort` we should  also try to, but I think the last commit is from 7 months ago, so maybe the PR will be stalled for a while.

Reply:
I'm not sure what to do here. natsorted gives a stupid error, but it feels slightly silly adding a pointless assert just to stop mypy complaining.

tuxemon/map_loader.py:395: error: Item "float" of "Union[str, bytes, float, int, PurePath, None, Iterable[Union[str, bytes, float, int, PurePath, None, Iterable[Any]]]]" has no attribute "startswith"
tuxemon/map_loader.py:395: error: Item "int" of "Union[str, bytes, float, int, PurePath, None, Iterable[Union[str, bytes, float, int, PurePath, None, Iterable[Any]]]]" has no attribute "startswith"
tuxemon/map_loader.py:395: error: Item "PurePath" of "Union[str, bytes, float, int, PurePath, None, Iterable[Union[str, bytes, float, int, PurePath, None, Iterable[Any]]]]" has no attribute "startswith"
tuxemon/map_loader.py:395: error: Item "None" of "Union[str, bytes, float, int, PurePath, None, Iterable[Union[str, bytes, float, int, PurePath, None, Iterable[Any]]]]" has no attribute "startswith"
tuxemon/map_loader.py:395: error: Item "Iterable[Union[str, bytes, float, int, PurePath, None, Iterable[Any]]]" of "Union[str, bytes, float, int, PurePath, None, Iterable[Union[str, bytes, float, int, PurePath, None, Iterable[Any]]]]" has no attribute "startswith"